src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,95 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * Parser class that makes sense of user command and executes appropriate actions.
+ */
+public class Parser {
+
+    /**
+     * Parses the user command and executes appropriate methods.
+     * @param command user input
+     * @param ui Ui object that deals with user interaction
+     * @param tasks TaskList object that stores all the tasks
+     * @param storage Storage object that saves tasks into a file
+     * @throws DukeException if command cannot be parsed
+     * @throws IOException if file cannot be opened.
+     */
+    public static void parseAndExecute(String command, Ui ui, TaskList tasks, Storage storage)
+            throws DukeException, IOException {
+        if (command.equals("list")) {
+            ui.printList(tasks);
+        } else {
+            if (command.indexOf(" ") == -1) {
+                throw new DukeException("Incorrect Format of Command");
+            }
+            String identifier = command.substring(0, command.indexOf(" "));
+            String task = command.substring(command.indexOf(" ") + 1);
+            String[] arr = task.split("[/]");
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd HHmm");
+            DateTimeFormatter formatter2 = DateTimeFormatter.ofPattern("HHmm");
```

Comment:
I've noticed that you keep using these 2 lines in multiple places in your code. Perhaps we could abstract this out somewhere, like in a utility/helper class?